alternative_backend/boards/serializers.py

```diff
@@ -1,27 +1,104 @@
 from rest_framework import serializers
+from rest_framework.validators import UniqueTogetherValidator
+from alternative_backend.exceptions import AppException
+from alternative_backend.settings import BARCODE_LENGHT
+from workers.models import Worker
+from stations.models import Station
+from orders.models import SendedBoard
 from .models import BoardCompany, BoardModel, Board, BoardScan
 
 
-class BoardCompanySerializer(serializers.HyperlinkedModelSerializer):
-	class Meta:
-		model = BoardCompany
-		fields = ('id', 'description', 'company_name')
 
+class BoardCompanySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = BoardCompany
+        fields = ('id', 'description', 'name', 'code')
 
-class BoardModelSerializer(serializers.HyperlinkedModelSerializer):
-	class Meta:
-		model = BoardModel
-		fields = ('id', 'description', 'year', 'company')
 
+class BoardModelSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = BoardModel
+        fields = ('id', 'description', 'year', 'company')
 
-class BoardSerializer(serializers.HyperlinkedModelSerializer):
-	class Meta:
-		model = Board
-		fields = ('id', 'model', 'barcode', 'company')
 
+class BoardSerializer(serializers.ModelSerializer):
+    def is_valid(self, raise_exception=False):
+        try:
+            model_code = int(str(self.initial_data['barcode'])[2:4])
+            company_code = int(str(self.initial_data['barcode'])[4:6])
+            model_id = BoardModel.objects.get(code=model_code).id
+            company_id = BoardCompany.objects.get(code=company_code).id
+        except:
```

Comment:
Zbyt szeroki zakres wyjątku. Zawsze ale to zawsze deklaruj jaki wyjątek chcesz obsłużyć. Bo jeśli wydarzy się coś nieprzewidzianego to nawet nie będziesz wiedział w czym problem.

```python
try:
   # rzeczy
except (KeyError, DbError, JakistamError):
   return false
```

jesli kod wysypie sie z powodow innych niz te 3 wymienione to przynajmniej dowiesz się, że coś jest nie tak.